libscanbuild/intercept.py

```diff
@@ -154,6 +154,19 @@ def intercept_compiler_wrapper(_, execution):
         logging.warning(message_prefix, 'io problem')
 
 
+def expand_cmd_with_response_files(cmd):
+    for param in cmd[:]:
+        if param[:1] == '@' and os.path.isfile(param[1:]):
+            # can throw but intercap_compiler_wrapper should catch
+            with open(param[1:], 'r') as f:
+                input_buffer = f.read()
+                idx = cmd.index(param)
+                del cmd[idx]
```

Comment:
Can we do the transformation without the deletion-insertion? I was thinking about a list comprehension solution like `[from_respone_file(arg) for arg in args if is_response_file(arg)]`. Where the `from_response_file` is the method which takes '@filename' and return the content (see previous comment). And the `is_response_file` checks the `@` prefix and the file existence. These methods can declared inside the method, but documented.